src/flags.rs

```diff
@@ -36,11 +37,75 @@ Environment variables:
   )
 }
 
-// Parses flags for deno. This does not do v8_set_flags() - call that separately.
+// Checks provided arguments for known options and sets appropriate Deno flags for them.
+// Unknown options are returned for further use.
+fn set_recognized_flags(
```

Comment:
Please add to the comment the two caveats of this approach: 
1. This assumes that privileged flags do not accept parameters `deno --foo bar`. This assumption is currently valid. But if it were to change in the future, this parsing technique would need to be modified. I _think_ we want to keep the privileged flags minimal - so having this restriction is maybe a good thing.
2. Misspelled flags will be forwarded to user code - e.g. `--allow-ne` would not cause an error. I also think this is ok because missing any of the privileged flags is not destructive. Userland flag parsing would catch these errors.